File: app/src/main/java/com/example/sarada/moviereviews/ReviewActivity.java
package com.example.sarada.moviereviews;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.Toast;

import com.example.sarada.moviereviews.models.Review;
import com.example.sarada.moviereviews.models.ReviewResponse;

import java.util.ArrayList;
import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class ReviewActivity extends AppCompatActivity {

    RecyclerView recyclerView2;
    int movie_id;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_review);
        recyclerView2 = findViewById(R.id.recycler_view2);
        movie_id = getIntent().getExtras().getInt("movieId");
        ReviewActivity.this.setTitle("Reviews of \n"+getIntent().getExtras().getString("movieName"));
        Toast.makeText(ReviewActivity.this,getIntent().getExtras().getString("movieName"),Toast.LENGTH_SHORT).show();
        initViews1();
    }

    private void initViews1(){
        List<Review> reviewList = new ArrayList<>();
        ReviewAdapter reviewAdapter = new ReviewAdapter(this, reviewList);
        RecyclerView.LayoutManager layoutManager1 = new LinearLayoutManager(getApplicationContext());
        recyclerView2.setLayoutManager(layoutManager1);
        recyclerView2.setAdapter(reviewAdapter);
        reviewAdapter.notifyDataSetChanged();
        loadJSON1();
    }

    private void loadJSON1(){

        try{
            if (BuildConfig.THE_MOVIE_DB_API_TOKEN.isEmpty()){
                Toast.makeText(getApplicationContext(), "Please obtain your API Key from themoviedb.org", Toast.LENGTH_SHORT).show();
                return;
            }
            RetrofitMake Client = new RetrofitMake();
            RetrofitQuery apiService = RetrofitMake.getClient().create(RetrofitQuery.class);
            Call<ReviewResponse> call = apiService.getMovieReviews(movie_id, BuildConfig.THE_MOVIE_DB_API_TOKEN);
            call.enqueue(new Callback<ReviewResponse>() {
                @Override
                public void onResponse(Call<ReviewResponse> call, Response<ReviewResponse> response) {
                    List<Review> review = response.body().getResults();
                    recyclerView2.setAdapter(new ReviewAdapter(getApplicationContext(), review));
                    recyclerView2.smoothScrollToPosition(0);
                }

                @Override
                public void onFailure(Call<ReviewResponse> call, Throwable t) {
                    Log.d("Error", t.getMessage());
                    Toast.makeText(ReviewActivity.this, "Error fetching trailer data", Toast.LENGTH_SHORT).show();

                }
            });

        }catch (Exception e){
            Log.d("Error", e.getMessage());
            Toast.makeText(this, e.toString(), Toast.LENGTH_SHORT).show();
        }
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.review_menu, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch(item.getItemId())
        {
            case R.id.home_item:
                Intent intent = new Intent(this, MainActivity.class);
                startActivity(intent);
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }
}

File: app/src/main/java/com/example/sarada/moviereviews/ReviewActivity.kt
package com.example.sarada.moviereviews

import com.example.sarada.moviereviews.RetrofitMake.client
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.RecyclerView
import android.os.Bundle
import com.example.sarada.moviereviews.R
import android.widget.Toast
import com.example.sarada.moviereviews.models.Review
import com.example.sarada.moviereviews.ReviewAdapter
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.sarada.moviereviews.RetrofitMake
import com.example.sarada.moviereviews.RetrofitQuery
import com.example.sarada.moviereviews.models.ReviewResponse
import android.content.Intent
import android.util.Log
import android.view.Menu
import android.view.MenuItem
import com.example.sarada.moviereviews.MainActivity
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import java.lang.Exception
import java.util.ArrayList

class ReviewActivity : AppCompatActivity() {
    var recyclerView2: RecyclerView? = null
    var movie_id = 0
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_review)
        recyclerView2 = findViewById(R.id.recycler_view2)
        movie_id = intent.extras!!.getInt("movieId")
        this@ReviewActivity.title = """
            Reviews of 
            ${intent.extras!!.getString("movieName")}
            """.trimIndent()
        Toast.makeText(
            this@ReviewActivity,
            intent.extras!!.getString("movieName"),
            Toast.LENGTH_SHORT
        ).show()
        initViews1()
    }

    private fun initViews1() {
        val reviewList: List<Review> = ArrayList()
        val reviewAdapter = ReviewAdapter(this, reviewList)
        val layoutManager1: RecyclerView.LayoutManager = LinearLayoutManager(applicationContext)
        recyclerView2!!.layoutManager = layoutManager1
        recyclerView2!!.adapter = reviewAdapter
        reviewAdapter.notifyDataSetChanged()
        loadJSON1()
    }

    private fun loadJSON1() {
        try {
            if (BuildConfig.THE_MOVIE_DB_API_TOKEN.isEmpty()) {
                Toast.makeText(
                    applicationContext,
                    "Please obtain your API Key from themoviedb.org",
                    Toast.LENGTH_SHORT
                ).show()
                return
            }
            val apiService = client!!.create(
                RetrofitQuery::class.java
            )
            val call = apiService.getMovieReviews(movie_id, BuildConfig.THE_MOVIE_DB_API_TOKEN)
            call!!.enqueue(object : Callback<ReviewResponse?> {
                override fun onResponse(
                    call: Call<ReviewResponse?>?,
                    response: Response<ReviewResponse?>?
                ) {
                    val review = response?.body()?.results
                    recyclerView2!!.adapter = review?.let { ReviewAdapter(applicationContext, it) }
                    recyclerView2!!.smoothScrollToPosition(0)
                }

                override fun onFailure(call: Call<ReviewResponse?>, t: Throwable) {
                    Log.d("Error", t.message!!)
                    Toast.makeText(
                        this@ReviewActivity,
                        "Error fetching trailer data",
                        Toast.LENGTH_SHORT
                    ).show()
                }
            })
        } catch (e: Exception) {
            Log.d("Error", e.message!!)
            Toast.makeText(this, e.toString(), Toast.LENGTH_SHORT).show()
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.review_menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            R.id.home_item -> {
                val intent = Intent(this, MainActivity::class.java)
                startActivity(intent)
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }
}